app/src/main/java/com/protify/protifyapp/features/calendar/EventBreakdown.kt

```diff
@@ -0,0 +1,373 @@
+                        overflow = TextOverflow.Ellipsis
+                    )
+//                    Text(
+//                        text = ("${timeSlot.startTime} - ${timeSlot.endTime}"),
+//                        modifier = Modifier.padding(end = 4.dp),
+//                        style = MaterialTheme.typography.bodyMedium
+//                    )
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun TimeSlotLayer(
+        timeSlots: List<TimeSlot>,
+        scale: Int,
+        layer: Int,
+        context: Context,
+        uid: String,
+        day: String,
+        month: String,
+        year: String,
+        scrollState: ScrollState
+    ) {
+        val layerColor = when (layer) {
+            0 -> MaterialTheme.colorScheme.primary
+            1 -> MaterialTheme.colorScheme.secondary
+            else -> MaterialTheme.colorScheme.error
+        }
+        Box(
+            modifier = Modifier
+                .fillMaxWidth(1f)
+                .padding(horizontal = 8.dp)
+                .verticalScroll(scrollState)
+                .background(Color.Transparent),
+            contentAlignment = Alignment.TopEnd
+        ) {
+            Column(
+                modifier = Modifier.fillMaxWidth(0.8f),
+                horizontalAlignment = Alignment.End
+            ) {
+                val startTimeMinutes = (convertTimeToFloat(timeSlots[0].startTime)) * 60
+                Spacer(modifier = Modifier.height(((30 + startTimeMinutes) * scale).dp)) // top padding
+                val displayedEvents = mutableMapOf<String, Int>()
+                // sort time slots by start time
+                val timeSlots = timeSlots.sortedBy { convertTimeToFloat(it.startTime) }
+                for (i in timeSlots.indices) {
+                    val timeSlot = timeSlots[i]
+
+                    val nextEventStartTime = if (i < timeSlots.size - 1) convertTimeToFloat(timeSlots[i + 1].startTime) else 24f
+                    val currentEventEndTime = convertTimeToFloat(timeSlot.endTime)
+                    val differenceInMinutes = (nextEventStartTime - currentEventEndTime) * 60
+
+                    val overlappingEvents = timeSlots.filter{ checkOverlap(
+                        Event().apply { startTime = it.startTime; endTime = it.endTime },
+                        Event().apply { startTime = timeSlot.startTime; endTime = timeSlot.endTime }
+                    )}
+
+                    if (overlappingEvents.size > 1) {
+                        Row(
+                            modifier = Modifier.fillMaxWidth(),
+                            horizontalArrangement = Arrangement.Start
+                        ) {
+                            for (timeSlot in overlappingEvents) {
+                                val count = displayedEvents.getOrDefault(timeSlot.id, 0)
+                                if (count < overlappingEvents.size && !displayedEvents.containsKey(timeSlot.id)) {
```

Comment:
!displayedEvents.containsKey(timeSlot.id) is critital to make sure events aren't put on the schedule more than once